liferay-workspace/modules/employee-common/src/main/java/com/emp/common/methods/EmployeeCommonServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package com.emp.common.methods;
+
+import java.util.List;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+import com.emp.common.constants.EmployeeCommonConstant;
+import com.employee.model.Employee;
+import com.employee.model.EmployeeTable;
+import com.employee.service.EmployeeLocalService;
+import com.liferay.petra.sql.dsl.DSLQueryFactoryUtil;
+import com.liferay.petra.sql.dsl.expression.Predicate;
+import com.liferay.petra.sql.dsl.query.DSLQuery;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.GetterUtil;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+@Component(immediate = true, service = EmployeeCommonService.class)
+public class EmployeeCommonServiceImpl implements EmployeeCommonService {
+
+	@Reference
+	private EmployeeLocalService employeeLocalService;
+
+	@Override
+	public List<Employee> getEmployeeList(String searchString, Long userId) {
+
+		DSLQuery dslQuery;
+		if (Validator.isNotNull(searchString)) {
+			String search = "%" + searchString + "%";
+			Long empId = null;
+			try {
+				empId = Long.parseLong(searchString);
+			} catch (Exception e) {
+				log.error(e);
+			}
+			Predicate userPredict = Predicate.withParentheses((EmployeeTable.INSTANCE.employeeId.eq(empId))
+					.or(EmployeeTable.INSTANCE.firstName.like(search)).or(EmployeeTable.INSTANCE.lastName.like(search))
+					.or(EmployeeTable.INSTANCE.mobileNumber.like(search))
+					.or(EmployeeTable.INSTANCE.emailAddress.like(search)));
+			dslQuery = DSLQueryFactoryUtil.select().from(EmployeeTable.INSTANCE)
+					.where((EmployeeTable.INSTANCE.createdby.eq(userId)).and(userPredict));
+
+		} else {
+			dslQuery = DSLQueryFactoryUtil.select().from(EmployeeTable.INSTANCE)
+					.where(EmployeeTable.INSTANCE.createdby.eq(userId));
+		}
+		List<Employee> employeeList = employeeLocalService.dslQuery(dslQuery);
```

Comment:
You can directly return employeeLocalService.dslQuery(dslQuery); instead of storing it into List.